### PR TITLE
Further cleanup of UBL

### PR DESCRIPTION
**Pull request by thinkyhead**
Rebase #6501 and apply more cleanup…

- Fix a bug in `smart_fill_mesh`
- Reduce code size of `smart_fill_mesh`
- Use 'PSTR' for `::debug` methods' title argument
- Fix formatting, apply coding standards

[Concise diff](https://github.com/MarlinFirmware/Marlin/pull/6552/files?w=1)

### CODE REVIEW

**Comment by Bob-the-Kuhn** [comment]
In UBL_G29.cpp, please replaces all the X_MIN_POS ... travel limits with the UBL_MESH_MIN_X ... versions.

**Comment by thinkyhead** [comment]
How about in G26?

File: Marlin/ubl_G29.cpp [hunk 1]
```diff
@@ -812,94 +776,96 @@
...
-              SERIAL_PROTOCOL_F( x_tmp, 7);
-              SERIAL_ECHOPGM(",");
-              SERIAL_PROTOCOL_F( y_tmp, 7);
-              SERIAL_ECHOPGM(",");
-              SERIAL_PROTOCOL_F( z_tmp, 7);
-              SERIAL_ECHOPGM("]   ---> ");
-              safe_delay(20);
+    for (uint8_t i = 0; i < GRID_MAX_POINTS_X; i++) {
+      for (uint8_t j = 0; j < GRID_MAX_POINTS_Y; j++) {
+        float x_tmp = pgm_read_float(ubl.mesh_index_to_xpos[i]),
```
Comment (bgort) [inline Marlin/ubl_G29.cpp]: Need an ampersand here, I believe.

`float x_tmp = pgm_read_float(ubl.mesh_index_to_xpos[i]),` 
should be 
`float x_tmp = pgm_read_float(&ubl.mesh_index_to_xpos[i]),`

Same with the next line.

Comment (thinkyhead) [reply Marlin/ubl_G29.cpp]: !!!

Comment (Roxy-3D) [reply Marlin/ubl_G29.cpp]: Agreed!

**Comment by Roxy-3D** [comment]
> How about in G26?

Yes...  for now...  that is probably the right thing to do.   But I suspect G26 has value to the Bi-Linear and original Mesh bed leveling.   In which case, using the UBL constants is not the 'right' thing to do...

**Comment by thinkyhead** [comment]
If it's "right" for UBL then it's ok. It should be self-consistent.

**Comment by thinkyhead** [comment]
Compare your smart fill with the bilinear one, inherited from @jcrocholl who inherited it from… who knows where?

https://github.com/MarlinFirmware/Marlin/blob/RCBugFix/Marlin/Marlin_main.cpp#L2648-L2684

**Comment by Roxy-3D** [comment]
> Wait, is this correct?
> The first three are squared, but the three others are not!

Good catch!  This is wrong....   These are xy, xz and yz terms in the polynomial.   They are not x^2, y^2 or z^2 terms.   The code will **NOT** work like this.

```cpp
-  lsf->x2bar = lsf->x2bar / N - lsf->xbar * lsf->xbar;
 -  lsf->y2bar = lsf->y2bar / N - lsf->ybar * lsf->ybar;
 -  lsf->z2bar = lsf->z2bar / N - lsf->zbar * lsf->zbar;
 -  lsf->xybar = lsf->xybar / N - lsf->xbar * lsf->ybar;
 -  lsf->yzbar = lsf->yzbar / N - lsf->ybar * lsf->zbar;
 -  lsf->xzbar = lsf->xzbar / N - lsf->xbar * lsf->zbar;
 +  lsf->x2bar = lsf->x2bar / N - sq(lsf->xbar);
 +  lsf->y2bar = lsf->y2bar / N - sq(lsf->ybar);
 +  lsf->z2bar = lsf->z2bar / N - sq(lsf->zbar);
 +  lsf->xybar = lsf->xybar / N - sq(lsf->xbar);
 +  lsf->yzbar = lsf->yzbar / N - sq(lsf->ybar);
 +  lsf->xzbar = lsf->xzbar / N - sq(lsf->xbar);
```

**Pull request merged**